infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironment.java

```diff
@@ -201,6 +207,14 @@ public void addPod(Pod pod) {
     return ImmutableMap.copyOf(injectablePods);
   }
 
+  public Map<String, String> getWorkspaceDeploymentLabels() {
```

Comment:
I think it makes sense to consider alternative way:
1. Provisioner - put annotation/label from WS to environment attributes
2. podMerger - get environment the corresponding enviroment attributes and propagate them to deployment object. 

In that way, it would not be too official and would not have a dedicated place in K8sEnvironment.